### PR TITLE
feat: Extension to Foundation library (vendor/darwin)

### PR DESCRIPTION
**Pull request by dvrd**
# Changes
* Adds `NSColor`, `NSEvent` & `NSScreen`
* Extends `NSApplication` to receive and send events
* Adds `NSRunLoopMode` constants

### CODE REVIEW

File: vendor/darwin/Foundation/objc.odin [hunk 1]
```diff
@@ -79,3 +77,14 @@ objc_class_internals :: struct {
...
+RunLoopMode :: ^String
+
+@(link_prefix="NS")
+foreign Foundation {
+	CommonRunLoopMode: RunLoopMode 
+	DefaultRunLoopMode: RunLoopMode
+	EventTrackingRunLoopMode: RunLoopMode
+	ModalPanelRunLoopMode: RunLoopMode
+	TrackingRunLoopMode: RunLoopMode
+}
```
Comment (Lperlind) [inline vendor/darwin/Foundation/objc.odin:90]: This should probably go into NSApplication.odin

File: vendor/darwin/Foundation/NSWindow.odin [hunk 2]
```diff
@@ -589,6 +594,10 @@ View :: struct {using _: Responder}
 View_initWithFrame :: proc "c" (self: ^View, frame: Rect) -> ^View {
 	return msgSend(^View, self, "initWithFrame:", frame)
 }
+@(objc_type=View, objc_name="bounds")
+View_bounds :: proc "c" (self: ^View) -> ^Rect {
+	return msgSend(^Rect, self, "bounds")
```
Comment (Lperlind) [inline vendor/darwin/Foundation/NSWindow.odin:599]: Bounds does not return a pointer to Rect.
https://developer.apple.com/documentation/appkit/nsview/1483817-bounds?language=objc

**Review by Lperlind — CHANGES_REQUESTED** [review]
Can you double check you've got all your return types are correct? For example you have made bounds return a pointer to a rect when it in fact does not and just returns the struct itself.

**Comment by gingerBill** [comment]
None of this was tested, was it?

**Comment by dvrd** [comment]
> None of this was tested, was it?

@gingerBill would you mind pointing to reference of tests regarding darwin Foundation inside the codebase?

**Comment by gingerBill** [comment]
You didn't test the code you wrote.

**Pull request closed without merging**